lib/src/revset.rs

```diff
@@ -683,14 +683,24 @@ static BUILTIN_FUNCTION_MAP: Lazy<HashMap<&'static str, RevsetFunction>> = Lazy:
     // code completion inside macro is quite restricted.
     let mut map: HashMap<&'static str, RevsetFunction> = HashMap::new();
     map.insert("parents", |diagnostics, function, context| {
-        let [arg] = function.expect_exact_arguments()?;
+        let ([arg], [depth_opt_arg]) = function.expect_arguments()?;
         let expression = lower_expression(diagnostics, arg, context)?;
-        Ok(expression.parents())
+        if let Some(depth_arg) = depth_opt_arg {
+            let depth = expect_literal("integer", depth_arg)?;
+            Ok(expression.ancestors_at(depth))
```

Comment:
nit: this can panic by `u64::MAX`. Since it's unlikely, and we clap overflow values in optimization pass, I think it's okay to clap overflow at `.ancestors`/`descendants_at()`. Returning error is also good, but might be a bit more verbose.